src/routes/profiles.py

```
@@ -1,5 +1,107 @@
+        first_name=first_name,
+        last_name=last_name,
+        gender=gender,
+        date_of_birth=date_of_birth,
+        info=info,
+        avatar=avatar
+    )
+
+    user_db = await db.execute(
+        select(UserModel)
+        .options(selectinload(UserModel.group), selectinload(UserModel.profile))
+        .where(UserModel.id == retrieved_user_id)
+    )
+    user_db = user_db.scalar_one_or_none()
+    if not user_db or not user_db.is_active:
+        raise HTTPException(
+            status_code=status.HTTP_401_UNAUTHORIZED,
+            detail="User not found or not active."
+        )
+
+    if user_id != retrieved_user_id and user_db.group.name != "admin":
+        raise HTTPException(
+            status_code=status.HTTP_403_FORBIDDEN,
+            detail="You don't have permission to edit this profile."
+        )
+    target_user = await db.execute(
+        select(UserModel)
+        .options(selectinload(UserModel.profile))
+        .where(UserModel.id == user_id)
+    )
+    target_user = target_user.scalar_one_or_none()
+    if not target_user:
+        HTTPException(status_code=401, detail="User not found or not active.")
+    user_db = target_user if user_id != retrieved_user_id else user_db
+    if user_db.profile:
+        raise HTTPException(
+            status_code=status.HTTP_400_BAD_REQUEST,
+            detail="User already has a profile."
+        )
+    file_bytes = await profile.avatar.read()
+    image_filename = f"avatars/{user_db.id}_{profile.avatar.filename}"
+    try:
+        await s3_client.upload_file(image_filename, file_bytes)
+        url = await s3_client.get_file_url(image_filename)
+    except (S3ConnectionError, S3FileUploadError):
+        raise HTTPException(
+            status_code=status.HTTP_500_INTERNAL_SERVER_ERROR,
+            detail="Failed to upload avatar. Please try again later."
+        )
+    gender = GenderEnum.MAN if profile.gender == "man" else GenderEnum.WOMAN
```

Minor: gender mapping `GenderEnum.MAN if profile.gender == "man" else GenderEnum.WOMAN` silently defaults to WOMAN for unexpected values. While your validators may prevent invalid values, consider an explicit mapping or raising an error if the value is unknown to avoid accidental incorrect mapping.